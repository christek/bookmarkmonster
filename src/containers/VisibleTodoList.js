// @flow

import { connect } from 'react-redux';

import { visibleTodosSelector } from '../selectors';
import TodoList from '../components/TodoList';

import type { State, Dispatch } from '../types';

const mapStateToProps = (state: State) => {
  return {
    todos: visibleTodosSelector(state)
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onTodoClick: id => {
      console.log('click');
    }
  };
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default connector(TodoList);
