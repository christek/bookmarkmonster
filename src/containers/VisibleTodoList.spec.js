// @flow

import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import VisibleTodoList from './VisibleTodoList';

const setup = (setupProps = {}) => {
  const store = configureStore()({
    todos: [
      {
        text: 'Test AddTodo',
        completed: false,
        id: 0
      },
      {
        text: 'Test AddTodo',
        completed: true,
        id: 1
      }
    ]
  });
  const wrapper = shallow(<VisibleTodoList store={store} />);

  return {
    store,
    wrapper
  };
};

describe('VisibleTodoList', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

});
