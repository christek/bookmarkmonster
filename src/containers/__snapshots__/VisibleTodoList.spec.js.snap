// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleTodoList renders without crashing 1`] = `ShallowWrapper {}`;
